### DISCUSSION TITLE
Adding extended attribute support?

### DISCUSSION THREAD

**Discussion by gchilds** [sparkle-project/Sparkle#2396] [General]
I was playing with the binary delta tool and was surprised to find it didn't support `xattr`s (extended
 attributes) on the before and after packages. Doesn't that exclude apps that contain scripts or third party frameworks? 
 I'm not sure how you're supposed to _not_ use extended attributes. 

How hard would it be to add xattr support? They're usually pretty small bits of metadata and could probably skip the diffing process altogether.

Thanks!

**Comment by zorgiepoo** [comment]
https://sparkle-project.org/documentation/delta-updates/

> Binary delta updates do not support and will reject the following metadata:
> * Extended attributes containing code signing information in either the old or new application. This can be resolved by properly structuring your application such that data and code are placed in the correct directories per [Code Signing in Depth guide](https://developer.apple.com/library/archive/technotes/tn2206/_index.html#//apple_ref/doc/uid/DTS40007919-CH1-TNTAG201). Only mach-o binaries should embed code signing information.

> Binary delta updates will pass but ignore or warn about the following metadata:
> * Extended attributes changes on files are not preserved and are just ignored.

Most likely your app is not structured correctly and has "code signing" extended attributes. Apple has a newer [Placing Content in a Bundle](https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle) guide that may be easier to digest.

Generally Sparkle ignores / doesn't make guarantees about preserving extended attributes. Extended attributes are flakey such that not all transit mechanisms guarantee they will be preserved. However extended attributes which contain code signing information (like incorrectly placing shell script files in directories expecting mach-o binaries) is very problematic if it gets lost. Hence, Sparkle does not support this and it's also illy advised.
